### PR TITLE
chore(types): ban explicit `any` and clean up violations

### PR DESCRIPTION
**Pull request by vzakharchenko**
## Summary
Enables `@typescript-eslint/no-explicit-any: error` and resolves all 187 violations across `src/` by either fixing (replacing with a concrete type) or suppressing with `// eslint-disable-next-line @typescript-eslint/no-explicit-any -- <reason>`.

Done in three commits on this branch:
1. **`catch (_: any)` → `unknown` with narrowing** — 14 catch clauses + 1 `Record<string, any>` (the original scope of the PR).
2. **`{ cause: parseError }` on rethrow in `Rovo.parseSqlQuery`** — satisfies `preserve-caught-error` that surfaced after typing.
3. **Enable `no-explicit-any` rule + clean up the rest** (this commit):
   - Fixes: `Array<any>` generics → `unknown[]`; `Record<string, any>` → `Record<string, unknown>`; `isQuery(obj: any)` / `isQueryError(error: any)` / `rethrowAsParsingError(error: any)` → `unknown` with inline shape casts; `params: any[]` / `rows: any[]` in the Drizzle proxy callback → `unknown[]`; `forgeDriver` reflection casts narrowed to `unknown[]` / `{ field?: unknown }`; `(oldModel as any)[fieldName]` → `Record<string, unknown>`; `calculateNewVersionValue(): any` → `Date | number`.
   - File-level suppress with reason: `lib/drizzle/extensions/additionalActions.ts` (Drizzle prototype augmentation), `utils/sqlUtils.ts` (Drizzle symbol-keyed reflection), `utils/cacheTableUtils.ts` (node-sql-parser AST traversal).
   - Per-line suppress with reason: AST-walking helpers in `Rovo`, the 9th `MySqlSelectBase` generic in `ForgeSQLQueryBuilder`, `MySqlRemoteDatabase<any>` in `ForgeSQLORM`, and Drizzle-boundary casts in `ForgeSQLCrudOperations` (`.values()`, `.set()`, `onDuplicateKeyUpdate`, `.select({ alias: column })`).

## Test plan
- [x] `npm run lint` — clean (no warnings, no errors)
- [x] `npm run build` (tsc) — clean
- [x] `npm test` — 726/726 passing
- [ ] Verify webtrigger error logs still include the SQL-message fields when an error bubbles from `@forge/sql`
- [ ] Spot-check that Drizzle .insert/.update/.delete still resolve the same generics in consumer code (Drizzle-boundary `as any` casts unchanged in behavior, only annotated)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * More consistent, safer error handling across queries, web triggers, caching and retry paths to surface clearer, non-sensitive error messages.

* **New Features**
  * Centralized error-message utilities used by APIs and web endpoints for clearer logs and HTTP responses.

* **Refactor**
  * Internal type tightening and standardized catch/rethrow patterns for more predictable runtime behavior.

* **Tests**
  * Added unit tests validating error-message extraction and normalization.

[Review Change Stack](https://app.coderabbit.ai/change-stack/forge-sql-orm/forge-sql-orm/pull/2291?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/core/Rovo.ts [hunk 1]
```diff
@@ -420,10 +420,9 @@ export class Rovo implements RovoIntegration {
     let ast;
     try {
       ast = parser.astify(sqlQuery);
-    } catch (parseError: any) {
-      throw new Error(
-        `SQL parsing error: ${parseError.message || "Invalid SQL syntax"}. Please check your query syntax.`,
-      );
+    } catch (parseError) {
+      const message = parseError instanceof Error ? parseError.message : "Invalid SQL syntax";
+      throw new Error(`SQL parsing error: ${message}. Please check your query syntax.`);
```
[the remark this answers is not in the document]
Comment (vzakharchenko) [reply src/core/Rovo.ts]: Addressed in 6a9f4bb (cause attached) and in 175d7382 (the implementation now uses extractSqlErrorMessage / getErrorMessage helpers that preserve detail for non-Error throws via String() coercion). Resolving.

File: src/core/ForgeSQLORM.ts [hunk 2]
```diff
@@ -274,13 +274,14 @@ class ForgeSQLORMImpl implements ForgeSqlOperation {
...
+        } catch (e) {
+          const err = e instanceof Error ? e : undefined;
           // eslint-disable-next-line no-console
           console.error(
             "[ForgeSQLORM][executeWithMetadata] Failed to run onMetadata callback",
             {
-              errorMessage: e?.message,
-              errorStack: e?.stack,
+              errorMessage: err?.message,
+              errorStack: err?.stack,
```
[the remark this answers is not in the document]
Comment (vzakharchenko) [reply src/core/ForgeSQLORM.ts]: Addressed in 175d7382. ForgeSQLORM.executeWithMetadata now uses getErrorMessage(e) for the message field (which preserves String(e) for non-Error throws) and only conditions e.stack on instanceof Error since stack is Error-specific. The original error e is still passed as the last argument to console.error for full context.

File: src/core/ForgeSQLCrudOperations.ts [hunk 3]
```diff
@@ -427,19 +434,18 @@ export class ForgeSQLCrudOperations implements VerioningModificationForgeSQL {
...
    * Calculates new version value based on field type.
    */
-  private calculateNewVersionValue(fieldType: string, currentVersion: unknown): any {
+  private calculateNewVersionValue(fieldType: string, currentVersion: unknown): Date | number {
     const dateTimeTypes = ["datetime", "timestamp"];
-    return dateTimeTypes.includes(fieldType)
-      ? new Date()
-      : (((currentVersion as number) + 1) as any);
+    return dateTimeTypes.includes(fieldType) ? new Date() : (currentVersion as number) + 1;
   }
```
[the remark this answers is not in the document]
Comment (vzakharchenko) [reply src/core/ForgeSQLCrudOperations.ts:452]: Addressed in 175d7382. calculateNewVersionValue now coerces currentVersion through Number(), so DECIMAL columns returned as strings increment correctly (Number("5") + 1 === 6 instead of "5" + 1 === "51"). Non-numeric inputs still propagate as NaN, matching the prior behavior the existing versioning-edge-case tests rely on (undefined / invalid version). A dedicated decimal test is a fair follow-up but not adding it in this PR to keep the scope focused on the any cleanup.
